features/google/src/model/importers/activities-importer-html.js

```diff
@@ -10,94 +13,22 @@ class ActivityHtmlParser {
     }
 
     _extractDate(text) {
-        // Ignoring the timestamp for now - we don't need hourly accuracy at the
-        // time of writing this.
-
         // Known formats:
         // 3 Feb 2001, 16:05:06 UTC
         // Feb 3, 2001, 4:05:06 PM UTC
         // 03.02.2001, 16:05:06 UTC
 
-        let date = new Date(text);
-
-        if (date && date !== null && date.toString() !== "Invalid Date") {
-            return date;
-        }
-
-        date = null;
-
-        const datePatterns = [
-            /([A-Z][a-z][a-z]) ([0-9]{1,2}), ([0-9]{4,4}), ([0-9:]{7,8} [A|P]M) ([A-Z]{3,4})/,
-            /([0-9]{1,2}) ([A-Z][a-z][a-z]) ([0-9]{4,4}), ([0-9:]{8,8}) ([A-Z]{3,4})/,
-            /([0-9]{1,2}).([0-9]{1,2}).([0-9]{4,4}), ([0-9:]{8,8}) ([A-Z]{3,4})/,
-        ];
-
-        // Add +1 to the index because the entire matched string will be at index 0 in match.
-        const dateMaps = [
-            {
-                day: 2,
-                month: 1,
-                year: 3,
-                time: 4,
-                timezone: 5,
-            },
-            {
-                day: 1,
-                month: 2,
-                year: 3,
-                time: 4,
-                timezone: 5,
-            },
-            {
-                day: 1,
-                month: 2,
-                year: 3,
-                time: 4,
-                timezone: 5,
-            },
-        ];
-
-        const month2Name = {
-            "01": "Jan",
-            1: "Jan",
-            "02": "Feb",
-            2: "Feb",
-            "03": "Mar",
-            3: "Mar",
-            "04": "Apr",
-            0: "Apr",
-            "05": "May",
-            5: "May",
-            "06": "Jun",
-            6: "Jun",
-            "07": "Jul",
-            7: "Jul",
-            "08": "Aug",
-            8: "Aug",
-            "09": "Sep",
-            9: "Sep",
-            10: "Oct",
-            11: "Nov",
-            12: "Dec",
-        };
-
-        for (let index = 0; index < datePatterns.length; index++) {
-            const match = text.match(datePatterns[index]);
-            if (match && match.length > 0) {
-                const day = match[dateMaps[index]["day"]];
-                let month = match[dateMaps[index]["month"]];
-                const year = match[dateMaps[index]["year"]];
-
-                if (!isNaN(parseInt(month))) {
-                    month = month2Name[month];
-                }
-
-                const dateString = `${day} ${month} ${year}`;
-                date = new Date(dateString);
+        // Replace named time offsets with UTC
+        for (const [key, value] of Object.entries(namedOffsets)) {
```

Comment:
Do we still need this? Doesn't `dayjs` support named timezones?

Reply:
It doesn't 😞

Reply:
If we introduce a third-party library for parsing the date, is there one that does support named timezones? [Luxon](https://moment.github.io/luxon/#/parsing) looks promising but I didn't try it out yet.

Reply:
I took a look at it. But they have this mention on their website:

----
[Ad-hoc parsing](https://moment.github.io/luxon/#/parsing?id=ad-hoc-parsing)
[Consider alternatives](https://moment.github.io/luxon/#/parsing?id=consider-alternatives)
You generally shouldn't use Luxon to parse arbitrarily formatted date strings:

If the string was generated by a computer for programmatic access, use a standard format like ISO 8601. Then you can parse it using DateTime.fromISO.
If the string is typed out by a human, it may not conform to the format you specify when asking Luxon to parse it. Luxon is quite strict about the format matching the string exactly.
Sometimes, though, you get a string from some legacy system in some terrible ad-hoc format and you need to parse it.
----

Adhoc parsing is what I wanted to have, to eliminate the regexes.

Reply:
Just to clarify, I am using dayjs, just because the date parsing is better than Date.parse, and so we will cover more cases. It also has the option of i18n, but I am not using that feature yet. I don't know how to take advantage of it.

Reply:
Also just to clarify, the browser's ' Date'  supports named timezones. It does support just a few, however, and for compatibility reasons, since apparently named timezones are deprecated. So, PST will work, EEST will not.